### PR TITLE
Optimize Server: `extractCritical`

### PR DESCRIPTION
**Pull request by lukeed**
Made some optimizations to the `server.js` code, as per @tkh44's request. This has been tested against the included suite, and I used those as a poor-man's benchmark 😅 I'd highly suggest/encourage these changes are run thru the _real_ benchmark suite before considering.

**Changes**

* Hoisted the RegExp pattern. It was redeclared twice & both times were at run-time
* Installed `@arr/filter` for fastest possible filtering. This function mutates(!), so it should be (and was) used carefully.
* Used a `forEach` loop instead of `map().join('')` because raw string concatenation is faster than joining an array.
* Removed the string-casting for `ids` keys. To me, it looks like they are/will always be strings. I may be wrong on this -- please correct me!

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -1,31 +1,32 @@
+import filter from '@arr/filter'
```
Comment (tkh44) [inline src/server.js]: Let's drop the filter. I don't want anyone to accidentally use it when mutations matter.

Comment (lukeed) [inline src/server.js]: You sure? This was responsible for largest drop in test suite time.

Comment (tkh44) [inline src/server.js]: Oh really?

Comment (tkh44) [inline src/server.js]: Keep it.

Comment (lukeed) [inline src/server.js]: Lol ok. IMO isn't too bad to keep an eye out on future changes/PRs 👻

Comment (thangngoc89) [inline src/server.js]: Eh. Add a bunch of comments about this "feature" of @arr/filter :D

**Pull request merged**